src/github.com/couchbase/sync_gateway/db/index_changes.go

```diff
@@ -168,13 +117,19 @@ func (db *Database) VectorMultiChangesFeed(chans base.Set, options ChangesOption
 				if minEntry.Seq.TriggeredBy == 0 {
 					// Update the cumulative clock, and stick it on the entry.
 					cumulativeClock.SetMaxSequence(minEntry.Seq.vbNo, minEntry.Seq.Seq)
-					// TODO: Potential enhancement - for first iteration loop, always set hash on last entry.
-					continuousLastHash = db.calculateHashWhenNeeded(
+					// Force new hash generation for non-continuous changes feeds if this is the last entry to be sent - either
+					// because there are no more entries in the channel feeds, or we're going to hit the limit.
+					forceHash := false
+					if options.Continuous == false && (nextEntry == nil || options.Limit == 1) {
+						forceHash = true
```

Comment:
so I take it if `options.Limit == 1` then `nextEntry` would be non-nil, because there might be lots of other entries but we only care about the first.  

But what about the case where `options.Limit == 2`?  Will `nextEntry == nil` on it's own be a reliable indicator that there are no more entries in the channel?